### PR TITLE
Seperating the Timeline into Timeline and Animation

### PR DESCRIPTION
**Pull request by m-agour**
In this PR, a new structure for the animation module is introduced while maintaining the old API (except for `add_child_timeline` is changed to become `add_child_animation`)
There is still modifing the tests. Tutorials will be updated in another PR.

### CODE REVIEW

File: fury/animation/animation.py [hunk 1]
```diff
@@ -0,0 +1,1385 @@
...
+        ----------
+        property_name: str
+            The name of the property.
+        cbk_func: function
+            The function to be called whenever the animation is updated.
+        """
+        attrib = self._get_attribute_data(property_name)
+        attrib.get('callbacks', []).append(cbk_func)
+
+    def update_animation(self, t=None):
```
Comment (skoudoro) [inline fury/animation/animation.py:1268]: avoid 1 letter variable.  3 letters minimum. to update

**Review by skoudoro — COMMENTED** [review]
See below a quick review, I will look deeper tomorrow

**Comment by skoudoro** [comment]
All tests are failing, can you fix the issue in this PR?  see here: https://github.com/fury-gl/fury/actions/runs/3135809301/jobs/5091933852#step:12:433

this is due to the renaming of `add_child_timeline`

**Comment by m-agour** [comment]
> All tests are failing, can you fix the issue in this PR? see here: https://github.com/fury-gl/fury/actions/runs/3135809301/jobs/5091933852#step:12:433
> 
> this is due to the renaming of `add_child_timeline`

Should I change the `Timelines` into `Animations` as well, in glTF animation?

**Comment by skoudoro** [comment]
> @skoudoro Should I change the Timelines into Animations as well, in glTF animation?

No @xtanion will have to do that, it will avoid conflict for him.

Thank you for the update

File: docs/tutorials/05_animation/viz_robot_arm_animation.py [hunk 2]
```diff
@@ -43,15 +44,23 @@
...
-tl_main.add_child_timeline(tl_child)
-tl_child.add_child_timeline(tl_grand_child)
+# Creating animations
+main_arm_animation = Animation([main_arm, joint_1], length=2 * np.pi)
+child_arm_animation = Animation([sub_arm, joint_2])
+drill_animation = Animation(end)
+
+###############################################################################
+# Adding the main animation to the Timeline.
+timeline.add_child_animation(main_arm_animation)
```
Comment (filipinascimento) [inline docs/tutorials/05_animation/viz_robot_arm_animation.py:58]: This should be `timeline.add_animation`, correct?

**Review by filipinascimento — COMMENTED** [review]
Thanks, this one looks good. Please check other places that you still may be calling `timeline.add_child_animation`.

**Comment by Garyfallidis** [comment]
@m-agour please rebase.

**Comment by m-agour** [comment]
Hi @Garyfallidis  @skoudoro, The scipy problem has been fixed now.

**Comment by Garyfallidis** [comment]
Thank you @m-agour.

**Pull request merged**